[java] CheckSkipResult: false positve for a private method `void skip(int)` in a subclass of FilterInputStream

**Affects PMD Version:** 7.5.0, 7.4.0, 7.3.0 (worked back in 7.2.0)

**Rule:**

*CheckSkipResult*
https://docs.pmd-code.org/pmd-doc-7.5.0/pmd_rules_java_errorprone.html#checkskipresult

**Description:**
Reproducer for Apache Commons Compress:
```
git clone https://gitbox.apache.org/repos/asf/commons-compress.git
cd commons-compress.git
git checkout 287e94beb6a81ddf111403b4c56ab80f957308b5
```
Fails when running:
```
mvn pmd:check -Dcommons.pmd-impl.version=7.4.0
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.24.0:check (default-cli) on project commons-compress: PMD 7.4.0 has found 3 violations. For more details see: C:\Users\ggregory\git\a\commons-compress\target\pmd.xml -> [Help 1]
```
Passes when running:
```
mvn commons.pmd-impl.version 
```
For all callers of a custom method:
```java
    private void skip(final int bytes) throws IOException {
        // bytes cannot be more than 3 bytes
        if (bytes > 0) {
            readFully(fourBytesBuf, 0, bytes);
        }
    }
```

**Expected outcome:**

PMD reports 3 violations at lines:
```xml
<?xml version="1.0" encoding="iso-8859-1"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 https://pmd.github.io/schema/report_2_0_0.xsd" version="7.5.0" timestamp="2024-09-11T17:25:13.760">
<file name="C:\Users\ggregory\git\a\commons-compress\src\main\java\org\apache\commons\compress\archivers\cpio\CpioArchiveInputStream.java">
<violation beginline="334" endline="334" begincolumn="13" endcolumn="42" rule="CheckSkipResult" ruleset="Error Prone" package="org.apache.commons.compress.archivers.cpio" class="CpioArchiveInputStream" method="read" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.5.0/pmd_rules_java_errorprone.html#checkskipresult" priority="3">
Check the value returned by the skip() method of an InputStream to see if the requested number of bytes has been skipped.
</violation>
<violation beginline="424" endline="424" begincolumn="9" endcolumn="50" rule="CheckSkipResult" ruleset="Error Prone" package="org.apache.commons.compress.archivers.cpio" class="CpioArchiveInputStream" method="readNewEntry" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.5.0/pmd_rules_java_errorprone.html#checkskipresult" priority="3">
Check the value returned by the skip() method of an InputStream to see if the requested number of bytes has been skipped.
</violation>
<violation beginline="487" endline="487" begincolumn="9" endcolumn="50" rule="CheckSkipResult" ruleset="Error Prone" package="org.apache.commons.compress.archivers.cpio" class="CpioArchiveInputStream" method="readOldBinaryEntry" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.5.0/pmd_rules_java_errorprone.html#checkskipresult" priority="3">
Check the value returned by the skip() method of an InputStream to see if the requested number of bytes has been skipped.
</violation>
</file>
<suppressedviolation filename="C:\Users\ggregory\git\a\commons-compress\src\main\java\org\apache\commons\compress\changes\ChangeSetPerformer.java" suppressiontype="//nopmd" msg="This if statement could be combined with its parent" usermsg=" NOSONAR
"></suppressedviolation>
</pmd>
```
All 3 violations are false positives.

**Running PMD through:** *Maven*

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -24,6 +24,8 @@ This is a {{ site.pmd.release_type }} release.
   * [#5059](https://github.com/pmd/pmd/issues/5059): \[core] xml output doesn't escape CDATA inside its own CDATA
 * java
   * [#5190](https://github.com/pmd/pmd/issues/5190): \[java] NPE in type inference
+* java-errorprone
+  * [#5207](https://github.com/pmd/pmd/issues/5207): \[java] CheckSkipResult: false positve for a private method `void skip(int)` in a subclass of FilterInputStream
 
 ### 🚨 API Changes
 

--- a/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CheckSkipResultRule.java
+++ b/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CheckSkipResultRule.java
@@ -11,7 +11,7 @@ import net.sourceforge.pmd.lang.java.types.InvocationMatcher;
 
 public class CheckSkipResultRule extends AbstractJavaRulechainRule {
 
-    private static final InvocationMatcher SKIP_METHOD = InvocationMatcher.parse("java.io.InputStream#skip(_*)");
+    private static final InvocationMatcher SKIP_METHOD = InvocationMatcher.parse("java.io.InputStream#skip(long)");
 
     public CheckSkipResultRule() {
         super(ASTMethodCall.class);

--- a/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/CheckSkipResult.xml
+++ b/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/CheckSkipResult.xml
@@ -116,4 +116,22 @@ public class Foo {
             }
             ]]></code>
     </test-code>
+
+    <test-code>
+        <description>#5207 FP with skip overload on custom FileInputStream class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.FileInputStream;
+import java.io.IOException;
+
+public class Foo extends FileInputStream {
+    public void bar() {
+        skip(1);
+    }
+
+    public void skip(int n) throws IOException {
+    }
+}
+        ]]></code>
+    </test-code>
 </test-data>

